db/migrate/20160113120732_dedupe_upps.rb

```diff
@@ -0,0 +1,19 @@
+class DedupeUpps < ActiveRecord::Migration
+  def change
+    UserProjectPreference.transaction do
+      UserProjectPreference.connection.execute <<-SQL
+        DELETE FROM user_project_preferences
+        WHERE id IN (
+          SELECT id
+          FROM (
+            SELECT id, ROW_NUMBER() OVER (PARTITION BY user_id, project_id ORDER BY id ASC) AS row
+            FROM user_project_preferences
+          ) dupes
+          WHERE dupes.row > 1
+        );
+      SQL
+
+      add_index :user_project_preferences, [:user_id, :project_id], unique: true
```

Comment:
be good to get rid of any if we can, that project_id one is 187MB, user_is is 33MB